crates/bevy_asset/src/reflect.rs

```diff
@@ -0,0 +1,305 @@
+use std::any::{Any, TypeId};
+
+use bevy_ecs::world::World;
+use bevy_reflect::{FromReflect, FromType, Reflect, Uuid};
+
+use crate::{Asset, Assets, Handle, HandleId, HandleUntyped};
+
+/// Type data for the [`TypeRegistry`](bevy_reflect::TypeRegistry) used to operate on reflected [`Asset`]s.
+///
+/// This type provides similar methods to [`Assets<T>`] like `get`, `add` and `remove`, but can be used in situations where you don't know which asset type `T` you want
+/// until runtime.
+///
+/// [`ReflectAsset`] can be obtained via [`TypeRegistration::data`](bevy_reflect::TypeRegistration::data) if the asset was registered using [`register_asset_reflect`](crate::AddAsset::register_asset_reflect).
+#[derive(Clone)]
+pub struct ReflectAsset {
```

Comment:
~~Nit: Would it make more sense to call this `ReflectAssets` (or something else) to better indicate that this is the "reflected version" of `Assets<T>` and not just the asset `T`?~~

On second thought, what you have here might be better as users will actually be working with the type of the asset and not the type of `Assets<T>` 😅